libraries/pod/src/option.rs

```diff
@@ -76,32 +93,33 @@ impl<T: Nullable> From<T> for PodOption<T> {
     }
 }
 
-impl<T: Nullable> From<Option<T>> for PodOption<T> {
-    fn from(from: Option<T>) -> Self {
-        match from {
-            Some(value) => PodOption(value),
-            None => PodOption(T::default()),
+impl<T: Nullable> TryFrom<Option<T>> for PodOption<T> {
+    type Error = ProgramError;
+
+    fn try_from(value: Option<T>) -> Result<Self, Self::Error> {
+        match value {
+            Some(value) if value.is_some() => Ok(PodOption(value)),
+            None => Ok(PodOption(T::NONE)),
+            _ => Err(ProgramError::InvalidArgument),
```

Comment:
nit: this is totally fine, but how about making this exhaustive to be safe?
```suggestion
            Some(value) if value.is_none() => Err(ProgramError::InvalidArgument),
            Some(value) => Ok(PodOption(value)),
            None => Ok(PodOption(T::NONE)),
```

Reply:
Nice!